x-pack/plugins/triggers_actions_ui/server/data/routes/indices.ts

```diff
@@ -66,7 +66,22 @@ export function createIndicesRoute(logger: Logger, router: IRouter, baseRoute: s
       logger.warn(`route ${path} error getting indices from pattern "${pattern}": ${err.message}`);
     }
 
-    const result = { indices: uniqueCombined(aliases, indices, MAX_INDICES) };
+    let dataStreams: string[] = [];
+    try {
+      dataStreams = await getDataStreamsFromPattern(esClient, pattern);
```

Comment:
From a very brief and incomplete historic review of this code, I think we made a mistake in not making the two calls we were already making in parallel.  And now we're adding a 3rd call, in serial fashion.  If I wasn't concerned about the response time of this API, I wouldn't mind it this way, but I believe this is probably called quite often while users are typing into the "index" field in the relevant rule UX, like every character (hopefully with some debounce!).

Seems like we can parallelize this fairly easily; instead of awaiting on the promises from getXYZFromPatten(), capture the promise where we can do a Promise.all() on it or whatever.

Reply:
That makes sense, I will update this!

Reply:
Hi Patrick, this has been updated in this [commit](https://github.com/elastic/kibana/pull/137584/commits/99acbbb9358bc921e912c842ce6a5e346157942c)